web/app.js

```diff
@@ -2651,50 +2654,62 @@ function webViewerWheel(evt) {
   ) {
     // Only zoom the pages, not the entire viewer.
     evt.preventDefault();
+
     // NOTE: this check must be placed *after* preventDefault.
     if (zoomDisabledTimeout || document.visibilityState === "hidden") {
       return;
     }
 
-    // It is important that we query deltaMode before delta{X,Y}, so that
-    // Firefox doesn't switch to DOM_DELTA_PIXEL mode for compat with other
-    // browsers, see https://bugzilla.mozilla.org/show_bug.cgi?id=1392460.
-    const deltaMode = evt.deltaMode;
-    const delta = normalizeWheelEventDirection(evt);
     const previousScale = pdfViewer.currentScale;
+    if (isPinchToZoom && supportsPinchToZoom) {
+      // The following formula is a bit strange but it comes from:
+      // https://searchfox.org/mozilla-central/rev/d62c4c4d5547064487006a1506287da394b64724/widget/InputData.cpp#618-626
+      const scaleFactor = Math.exp(-evt.deltaY / 100);
+      if (scaleFactor < 1) {
+        PDFViewerApplication.zoomOut(null, scaleFactor);
+      } else {
```

Comment:
Given how the exponentiation works, it's theoretically possible that `scaleFactor === 1` if `evt.deltaY = 0` ever happens, in which case we'd get into trouble when trying to zoom. Hence you'll need the following here:
```suggestion
      const scaleFactor = Math.exp(-evt.deltaY / 100);
      if (scaleFactor < 1) {
        PDFViewerApplication.zoomOut(null, scaleFactor);
      } else if (scaleFactor > 1) {
```